host.go

```diff
@@ -268,33 +268,36 @@ func (h *gpbftRunner) BroadcastMessage(msg *gpbft.GMessage) error {
 
 var _ pubsub.ValidatorEx = (*gpbftRunner)(nil).validatePubsubMessage
 
-func (h *gpbftRunner) validatePubsubMessage(ctx context.Context, pID peer.ID,
-	msg *pubsub.Message) pubsub.ValidationResult {
+func (h *gpbftRunner) validatePubsubMessage(_ context.Context, _ peer.ID, msg *pubsub.Message) (_result pubsub.ValidationResult) {
+	start := time.Now()
+	defer func() {
+		recordValidationTime(start, _result)
+	}()
+
 	var gmsg gpbft.GMessage
-	err := gmsg.UnmarshalCBOR(bytes.NewReader(msg.Data))
-	if err != nil {
-		return pubsub.ValidationReject
+	if err := gmsg.UnmarshalCBOR(bytes.NewReader(msg.Data)); err != nil {
+		_result = pubsub.ValidationReject
```

Comment:
Just return this. Named return values are always accessible from deferred functions (and bare returns are a language feature that should never have been implemented).

Reply:
Or, per the go tour:

>  Naked return statements should be used only in short functions, as with the example shown here. They can harm readability in longer functions. 

Although I still think they should never have been added...

Reply:
Also note: the reason we generally use `_name` for named return values instead of just `name` is to make it clear that they shouldn't be set. 

The concrete problem is that local variables can shadow named return values.